### PR TITLE
PIM-9469: [Backport] API-1232: Be able to get list of attribute from the API

### PR DESCRIPTION
**Pull request by anaisbaunelemaire**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added legacy Behats               | 
| Added acceptance tests            | 
| Added integration tests           | 
| Changelog updated                 | OK
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) | 
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/AttributeRepository.php [hunk 1]
```diff
@@ -113,7 +121,7 @@ protected function validateSearchFilters(array $searchFilters): void
         if (empty($searchFilters)) {
             return;
         }
-        $availableSearchFilters = ['code'];
+        $availableSearchFilters = ['code', 'updated'];
```
Comment (pierrejolly) [inline src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/AttributeRepository.php:124]: What about `type` ?

Comment (anaisbaunelemaire) [reply src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/AttributeRepository.php:124]: It's done in this PR, which is not merged yet => https://github.com/akeneo/pim-community-dev/pull/12823/files

Comment (anaisbaunelemaire) [reply src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/AttributeRepository.php:124]: It's done in this pr => https://github.com/akeneo/pim-community-dev/pull/12823
I have to merge API-1232 backport before (tickets were not classified correctly)

Comment (pierrejolly) [reply src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/AttributeRepository.php:124]: In #12823 you totally remove this line :/

**Review by pierrejolly — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/AttributeRepository.php (hunk 1 above)
Comment (anaisbaunelemaire) [reply src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/AttributeRepository.php:124]: Nope, it's updated ;-)  => `$availableSearchFilters = array_keys($constraints);`

File: CHANGELOG-4.0.md [hunk 2]
```diff
@@ -188,7 +189,7 @@
 
 ## Bug fixes
 
-- PIM-9055: Allow not to drop an existing database during the install process
+- PIM-9260: do not use FPM memory_limit for CommandLauncher
```
Comment (ValentinMumble) [inline CHANGELOG-4.0.md:192]: Why changing this entry?

**Review by ValentinMumble — APPROVED** [review]

File: CHANGELOG-4.0.md (hunk 2 above)
Comment (anaisbaunelemaire) [reply CHANGELOG-4.0.md:192]: I had modified this line by mistake in a previous pr, and I am correcting this error here ;-)

File: tests/back/Pim/Structure/EndToEnd/Attribute/ExternalApi/ListAttributeEndToEnd.php [hunk 3]
```diff
@@ -144,6 +145,52 @@ public function testAttributeSearchByCode()
         $this->assertJsonStringEqualsJsonString($expected, $response->getContent());
     }
 
+    public function testAttributeSearchByUpdated()
+    {
+        /** @var Connection $connection */
+        $connection = $this->get('database_connection');
+        $affected = $connection->exec('UPDATE pim_catalog_attribute SET updated="2019-05-15 16:27:00" WHERE code="a_file"');
+        $this->assertEquals(1, $affected, 'There is more result as expected during test setup, the test will not work.');
```
Comment (ahocquard) [inline tests/back/Pim/Structure/EndToEnd/Attribute/ExternalApi/ListAttributeEndToEnd.php:153]:
```suggestion
        Assert::assertEquals(1, $affected, 'There is more result as expected during test setup, the test will not work.');
```

Not a real issue here, no need to fix it, but for the next time :)  Static method `assertEquals` should not be called with `$this` (not possible with stricter languages as Java).

**Review by ahocquard — APPROVED** [review]

File: tests/back/Pim/Structure/EndToEnd/Attribute/ExternalApi/ListAttributeEndToEnd.php (hunk 3 above)
Comment (anaisbaunelemaire) [reply tests/back/Pim/Structure/EndToEnd/Attribute/ExternalApi/ListAttributeEndToEnd.php:153]: Thanks for the explanation, Alexandre !

**Pull request merged**